src/hotspot/share/gc/g1/g1FullGCHeapRegionAttr.hpp

```diff
@@ -54,35 +54,22 @@ class G1FullGCHeapRegionAttr : public G1BiasedMappedArray<uint8_t> {
 
 public:
   void set_invalid(uint idx) { set_by_index(idx, Invalid); }
+  void set_compacted(uint idx) { set_by_index(idx, Compacted); }
+  void set_not_marked_through(uint idx) { set_by_index(idx, NotMarkedThrough); }
+  void set_not_compacted(uint idx) { set_by_index(idx, NotCompacted); }
 
-  void set_closed_archive(uint idx) { set_by_index(idx, ClosedArchive); }
-
-  bool is_closed_archive(HeapWord* obj) const {
-    assert(!is_invalid(obj), "not initialized yet");
-    return get_by_address(obj) == ClosedArchive;
-  }
-
-  void set_pinned(uint idx) { set_by_index(idx, Pinned); }
-
-  bool is_pinned_or_closed(HeapWord* obj) const {
+  bool is_not_marked_through(HeapWord* obj) const {
     assert(!is_invalid(obj), "not initialized yet");
-    return get_by_address(obj) >= Pinned;
+    return get_by_address(obj) == NotMarkedThrough;
   }
 
-  bool is_pinned(HeapWord* obj) const {
+  bool is_compacted(HeapWord* obj) const {
     assert(!is_invalid(obj), "not initialized yet");
-    return get_by_address(obj) == Pinned;
-  }
-
-  bool is_pinned(uint idx) const {
-    return get_by_index(idx) == Pinned;
+    return get_by_address(obj) == Compacted;
   }
 
-  void set_normal(uint idx) { set_by_index(idx, Normal); }
-
-  bool is_normal(HeapWord* obj) const {
-    assert(!is_invalid(obj), "not initialized yet");
-    return get_by_address(obj) == Normal;
+  bool is_compacted_or_marked_through(uint idx) const {
```

Comment:
Should this be rename to is_compacted_or_not_marked_through?
As the condition is indeed == Compacted || == NotMarkedThrough.

Reply:
Yes, that's true. It should actually be called `is_compacted_or_not_marked_through`.

Reply:
See the other comment from me about naming variants provided by now.